client/src/partials/reference_group/reference_group.js

```diff
@@ -102,21 +96,15 @@ angular.module('bhima.controllers')
 
     $scope.save.new = function () {
       var record = connect.clean(session.new);
-
       connect.post('reference_group', [record])
-      .then(function () {
-        messenger.success($translate.instant('REFERENCE_GROUP.SAVE_SUCCES'));        
-        record.reference = generateReference(); 
+      .then(function (res) {
+        messenger.success($translate.instant('REFERENCE_GROUP.SAVE_SUCCES'));  
+        record.id = res.data.insertId;               
         $scope.reference_groups.post(record);
         session.action = '';
         session.new = {};
       });
     };
 
-    function generateReference () {
```

Comment:
I am happy to see these functions being removed.  They were a terrible hack up until now. :)